### PR TITLE
Vendor typing._SpecialForm to fool typing._type_check

### PR DESCRIPTION
**Pull request by CharString**
Adds a local copy of _SpecialForm in our namespace, so
typing._type_check won't raise TypeError. (#964)

### CODE REVIEW

**Comment by JelleZijlstra** [comment]
This has severe merge conflicts because we just renamed src_py3 to src, could you rebase?

**Comment by CharString** [comment]
:) I was looking into that.

File: typing_extensions/src/typing_extensions.py [hunk 1]
```diff
@@ -2048,40 +2048,61 @@ def __eq__(self, other):
...
+            raise AttributeError(item)
+
+        def __mro_entries__(self, bases):
+            raise TypeError(f"Cannot subclass {self!r}")
 
-elif sys.version_info[:2] >= (3, 9):
-    class _SelfForm(typing._SpecialForm, _root=True):
         def __repr__(self):
-            return 'typing_extensions.' + self._name
+            return 'typing.' + self._name
```
Comment (Gobot1234) [inline typing_extensions/src/typing_extensions.py:2071]:
```suggestion
            return f'typing_extensions.{self._name}'
```

File: typing_extensions/src/typing_extensions.py [hunk 2]
```diff
@@ -2048,40 +2048,61 @@ def __eq__(self, other):
 
     TypeGuard = _TypeGuard(_root=True)
 
+if not hasattr(typing, "Self") and sys.version_info[:2] >= (3, 7):
+    # Vendored from cpython typing._SpecialFrom
```
Comment (JelleZijlstra) [inline typing_extensions/src/typing_extensions.py:2052]: It's a bit hard to follow the if chains here, but can't this just be together with the appropriate branch for 3.7-3.10 so we don't have to duplicate the conditions?

Comment (CharString) [reply typing_extensions/src/typing_extensions.py:2054]: Should I refactor the uses of typing._SpecialForm to use this vendored  verion too?

Comment (JelleZijlstra) [reply typing_extensions/src/typing_extensions.py:2054]: Let's change Self only for simplicity.

**Review by JelleZijlstra — APPROVED** [review]
